base/linalg/transpose.jl

```diff
@@ -0,0 +1,334 @@
+immutable Transpose{T,S<:AbstractMatrix} <: AbstractMatrix{T}
+    data::S
+    conjugated::Bool
+end
```

Comment:
I've been back and forth here. The arguments against a type parameter are

1) More types cases hence more methods to compile (though the field will make the methods longer so it's not necessarily more expensive)
2) The signatures become more verbose
3) I couldn't come up with an example where dispatch on conjugation was necessary, i.e. where the return type would depend on it. The closest thing I could come up with was something like `X'X` where it would be neat if the result was `Symmetric`/`Hermitian` but that would require dispatch on aliasing as well which is not supported. If you have an example where the return type would depend on conjugation then please let me know.